pkg/config/config_test.go

```diff
@@ -113,8 +114,12 @@ func TestConfig_BuildBlockAdapter(t *testing.T) {
 		testutil.Must(t, err)
 		adapter, err := factory.BuildBlockAdapter(ctx, nil, c)
 		testutil.Must(t, err)
-		if _, ok := adapter.(*local.Adapter); !ok {
-			t.Fatalf("expected a local block adapter, got something else instead")
+		metricsAdapter, ok := adapter.(*block.MetricsAdapter)
+		if !ok {
+			t.Fatalf("expected a metrics block adapter, got something else instead")
+		}
+		if _, ok := metricsAdapter.InnerAdapter().(*local.Adapter); !ok {
+			t.Fatalf("expected an gs block adapter, got something else instead")
```

Comment:
This can and should be a compile-time test (if it's important enough to do).  Just add a line
```go
var _ Adapter = &MetricsAdapter{}
```
to the source (not a test) and I think we're done.

Reply:
Okay, I'm really confused now I guess.  BuildBlockAdapter could return a MetricsAdapter but instead returns an Adapter, and here we test the type, right?  So my compile-time test is indeed unneeded.

So everything cool here except the error message is confusing:
```suggestion
		metricsAdapter, ok := adapter.(*block.MetricsAdapter)
		if !ok {
			t.Fatalf("expected a metrics block adapter, got something else instead")
		}
		if _, ok := metricsAdapter.InnerAdapter().(*local.Adapter); !ok {
			t.Fatalf("got %T expected a local block adapter", metricsAdapter.InnerAdapter())
```